packages/dds/pact-map/src/pactMap.ts

```diff
@@ -208,6 +208,22 @@ export class PactMap<T = unknown> extends SharedObject<IPactMapEvents> implement
 		return this.values.get(key)?.accepted?.value;
 	}
 
+	/**
+	 * {@inheritDoc IPactMap.getDetails}
+	 */
+	public getDetails(key: string): IAcceptedPactDetails<T> | undefined {
+		// Note: We return type `IAcceptedPactDetails` instead of `IAcceptedPact` since we may want to diverge
+		// the interfaces in the future.
+		const acceptedPact = this.values.get(key)?.accepted;
+		if (acceptedPact?.value === undefined) {
```

Comment:
It's possible that the value is explicitly `undefined` here (e.g. I proposed `undefined` and it was accepted at sequence number 30), in which case we should still return the details.
```suggestion
		if (acceptedPact === undefined) {
```

Reply:
I think it's good to return the sequence number for a deletion - otherwise clients will see an "accept" for the deletion but be unable to get the details (sequenceNumber) of that deletion.  But then I think this also means we can't trim the summary anymore, since that state is observable.

So I'd probably make the change I suggest here but then also remove the summary-trimming logic from summarizeCore.